### PR TITLE
88

### PR DESCRIPTION
**Pull request by victornoel**
This is for #88, see commit messages for details.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#105](https://github.com/artipie/http/issues/105) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/d119fe14-529e-4026-a549-45aa12426d13); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/http/rq/Multipart.java [hunk 1]
```diff
@@ -71,22 +68,23 @@
...
      * Ctor.
      *
-     * @param boundary Multipart body boundary
+     * @param headers Request headers.
      */
-    public Multipart(final String boundary) {
-        this(new ByteByByteSplit(boundary.getBytes(StandardCharsets.UTF_8)));
+    public Multipart(final Iterable<Map.Entry<String, String>> headers) {
+        this(() -> {
+            final Pattern pattern = Pattern.compile("boundary=(\\w+)");
```
Comment (Vatavuk) [inline src/main/java/com/artipie/http/rq/Multipart.java:78]: @victornoel let's extract this to a private static method to make it more readable of what are we doing here.

**Comment by victornoel** [comment]
@Vatavuk tank you, it is done

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@Vatavuk please make sure you will find at least 3 issues during next CR, thanks

**Comment by Vatavuk** [comment]
@sereshqua ok

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**